src/timedate/timedated.c

```diff
@@ -873,6 +873,11 @@ static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *erro
         if (!relative && utc <= 0)
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid absolute time");
 
+        /* when utc time is past systemd build date time*/
+       	/* convert epoch time to microseconds to compare with utc*/
+        if(!relative && utc <= ((int64_t)TIME_EPOCH * 1000000))
```

Comment:
use USEC_PER_SEC instead of "1000000"

also, space missing after `if`, i.e. follow coding style otherwise applied her.

Reply:
hmm, you only do this check if absolute times are specified. I am pretty sure we should also apply this if relative times are specified.

Please move this check down, right before the `bus_verify_polkit_async_full` and then change the check to `if (tv.tv_sec < TIME_EPOCH)`...

Reply:
done, made the change